datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2528,6 +2568,14 @@ impl Aggregate {
         })
     }
 
+    /// Get the output expressions.
+    fn output_expressions(&self) -> Result<Vec<Expr>> {
```

Comment:
We have been trying to remove copies from the optimizer (see https://github.com/apache/datafusion/issues/9637) and this API forces a copy.

I wonder if we could avoid copies by returning references here instead 🤔  I looked at the above code and it wasn't immediately clear we could do so

```suggestion
    fn output_expressions(&self) -> Result<Vec<&Expr>> {
```

I think then you could have like this perhaps

```rust
    pub(crate) fn columnized_output_exprs(&self) -> Result<Vec<(&Expr, Column)>> {
```

Reply:
We can't do this now because [grouping_set_to_exprlist](https://github.com/jonahgao/datafusion/blob/131d1a96e64e84a01b954d7f5658976cbb64e83a/datafusion/expr/src/logical_plan/plan.rs#L2573) returns a temporary `Vec<Expr>`.

We need to first remove the copies in `grouping_set_to_exprlist`, and also need to modify [`exprlist_to_fields`](https://github.com/jonahgao/datafusion/blob/131d1a96e64e84a01b954d7f5658976cbb64e83a/datafusion/expr/src/logical_plan/plan.rs#L2515). `exprlist_to_fields` is referenced in many places. Maybe we can try to do this in subsequent PRs.